src/App.tsx

```diff
@@ -14,11 +14,114 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isVisible: boolean,
+  goodsList: string[],
+};
+
+class App extends React.Component <{}, State> {
+  state: Readonly <State> = {
+    isVisible: false,
+    goodsList: [...goodsFromServer],
+  };
+
+  visabilityToggle = () => {
+    this.setState(state => ({
+      isVisible: !state.isVisible,
+    }));
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      goodsList: state.goodsList.reverse(),
+    }));
+  };
+
+  sortAlphabetically = () => {
+    this.setState(state => ({
+      goodsList: state.goodsList.sort((a, b) => a.localeCompare(b)),
+    }));
+  };
+
+  reset = () => {
+    this.setState({ goodsList: [...goodsFromServer] });
+  };
+
+  sortByLength = () => {
+    this.setState(state => ({
+      goodsList: state.goodsList.sort((a, b) => a.length - b.length),
+    }));
+  };
+
+  render() {
+    const { isVisible, goodsList } = this.state;
+
+    return (
+      <div className="is-fluid">
+        {
+          isVisible
+            ? (
+              <div className="box is-medium ">
+                <h1 className="title">Goods:</h1>
+                <ul className="Goods__list content is-medium">
+                  {goodsList.map(
+                    good => (
+                      <li key={good}>
+                        {good}
+                      </li>
+                    ),
+                  )}
+                  {' '}
+                </ul>
+                <button
```

Comment:
add one empty line between same level elements
```suggestion
                </ul>
                
                <button
```